### PR TITLE
New module: anndata/getsize module

### LINKED ISSUE #6923: new module: anndata/getsize
### Is there an existing module for this?

- [x] I have searched for the existing module

### Is there an open PR for this?

- [x] I have searched for existing PRs

### Is there an open issue for this?

- [x] I have searched for existing issues

### Further Information

https://anndata.readthedocs.io/en/latest/generated/anndata.AnnData.shape.html

### Are you going to work on this?

- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by LeonHafner**
Adding the module `anndata/getsize` which outputs the size of an anndata object.
This can then be used to add the size of the anndata object to its meta map:
`ANNDATA_GETSIZE.out.size.map{meta, txt -> [meta, txt.text.toInteger()]}` 

Currently there is no proper `H5AD` data on the `test-datasets` modules branch. For now I used the files from the `scdownstream` pipeline, but I will try to add proper test data to the modules asap.

## PR checklist

Closes #6923  

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`

### CODE REVIEW

File: modules/nf-core/anndata/getsize/templates/getsize.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+
+import platform
+
+import anndata as ad
+
+
```
Comment (SPPearce) [inline modules/nf-core/anndata/getsize/templates/getsize.py:7]:
```suggestion
```

File: modules/nf-core/anndata/getsize/templates/getsize.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+
```
Comment (SPPearce) [inline modules/nf-core/anndata/getsize/templates/getsize.py:2]: What is the copyright and authors for this code?

**Review by SPPearce — COMMENTED** [review]
Generally fine.
It would be good to put a h5ad file into the modules test-datasets though, are you able to do that?

File: modules/nf-core/anndata/getsize/main.nf [hunk 3]
```diff
@@ -0,0 +1,36 @@
+process ANNDATA_GETSIZE {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'oras://community.wave.seqera.io/library/anndata:0.10.9--d13580e4b297da7c':
```
Comment (SPPearce) [inline modules/nf-core/anndata/getsize/main.nf:7]: I think this link is wrong, sent you some instructions on slack.

File: modules/nf-core/anndata/getsize/templates/getsize.py (hunk 1 above)
Comment (LeonHafner) [reply modules/nf-core/anndata/getsize/templates/getsize.py:16]: prettier wants those empty lines

File: modules/nf-core/anndata/getsize/main.nf (hunk 3 above)
Comment (LeonHafner) [reply modules/nf-core/anndata/getsize/main.nf]: thanks, replaced the url

File: modules/nf-core/anndata/getsize/main.nf [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${prefix}.txt
+
+    cat <<-END_VERSIONS > versions.yml
+    ${task.process}:
+        python: \$(python3 -c 'import platform; print(platform.python_version())')
+        anndata: \$(python3 -c 'import anndata as ad; print(ad.__version__)')
```
Comment (SPPearce) [inline modules/nf-core/anndata/getsize/main.nf:33]:
```suggestion
        anndata: \$(python3 -c 'import anndata; print(anndata.__version__)')
```

File: modules/nf-core/anndata/getsize/templates/getsize.py (hunk 1 above)
Comment (SPPearce) [reply modules/nf-core/anndata/getsize/templates/getsize.py:16]: Hah, fine, prettier wins then!

**Review by SPPearce — APPROVED** [review]
Looks good to me now, thanks for putting the file into the modules testdata repo.

**Pull request merged**